test/parallel/test-assert-first-line.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+// Verify that asserting in the very first line produces the expected result.
+
+require('../common');
+const assert = require('assert');
+const { path } = require('../common/fixtures');
+
+assert.throws(
+  () => require(path('assert-first-line')),
+  {
+    name: 'AssertionError [ERR_ASSERTION]',
+    message: "The expression evaluated to a falsy value:\n\n  assert.ok('')\n"
```

Comment:
could we assert that `line 1` shows up somewhere in the stack trace, I imagine you already thought of this and it's too much of a hassle?

Reply:
I am not sure if I understand exactly what you request. The stack trace is not impacted by the code that is read here.

Reply:
Since this bug was only manifesting itself if an exception happened on the first line of the application, I was wondering if it's worthwhile asserting that this was the line in the stack trace that caused the error.

This is probably overkill, since the fixture name `assert-first-line` is pretty clear about what's being tested. No blockers from me.

Reply:
Since the fixture itself can only throw on line 1, I think it is fine to keep it as is :)